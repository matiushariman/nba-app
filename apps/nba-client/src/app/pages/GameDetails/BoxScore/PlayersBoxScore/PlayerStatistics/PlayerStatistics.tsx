import {
  getPlayingTime,
  getShootingPercentage,
} from './PlayersStatistics.utils';

import type { PlayersStatisticsProps } from './PlayersStatistics.types';
const PlayerStatistics = ({ statistics, played }: PlayersStatisticsProps) => {
  if (played === '0') {
    return (
      <td colSpan={20} className="pl-4 text-sm">
        DNP
      </td>
    );
  }

  const {
    minutes,
    fieldGoalsMade,
    fieldGoalsAttempted,
    fieldGoalsPercentage,
    threePointersMade,
    threePointersAttempted,
    threePointersPercentage,
    freeThrowsMade,
    freeThrowsAttempted,
    freeThrowsPercentage,
    reboundsOffensive,
    reboundsDefensive,
    reboundsTotal,
    assists,
    steals,
    blocks,
    turnovers,
    foulsPersonal,
    points,
    plusMinusPoints,
  } = statistics;

  return (
    <>
      <td align="right" className="text-sm">
        {getPlayingTime(minutes)}
      </td>
      <td align="right" className="text-sm">
        {fieldGoalsMade}
      </td>
      <td align="right" className="text-sm">
        {fieldGoalsAttempted}
      </td>
      <td align="right" className="text-sm">
        {getShootingPercentage(fieldGoalsPercentage)}
      </td>
      <td align="right" className="text-sm">
        {threePointersMade}
      </td>
      <td align="right" className="text-sm">
        {threePointersAttempted}
      </td>
      <td align="right" className="text-sm">
        {getShootingPercentage(threePointersPercentage)}
      </td>
      <td align="right" className="text-sm">
        {freeThrowsMade}
      </td>
      <td align="right" className="text-sm">
        {freeThrowsAttempted}
      </td>
      <td align="right" className="text-sm">
        {getShootingPercentage(freeThrowsPercentage)}
      </td>
      <td align="right" className="text-sm">
        {reboundsOffensive}
      </td>
      <td align="right" className="text-sm">
        {reboundsDefensive}
      </td>
      <td align="right" className="text-sm">
        {reboundsTotal}
      </td>
      <td align="right" className="text-sm">
        {assists}
      </td>
      <td align="right" className="text-sm">
        {steals}
      </td>
      <td align="right" className="text-sm">
        {blocks}
      </td>
      <td align="right" className="text-sm">
        {turnovers}
      </td>
      <td align="right" className="text-sm">
        {foulsPersonal}
      </td>
      <td align="right" className="text-sm">
        {points}
      </td>
      <td align="right" className="text-sm">
        {plusMinusPoints}
      </td>
    </>
  );
};

export default PlayerStatistics;
