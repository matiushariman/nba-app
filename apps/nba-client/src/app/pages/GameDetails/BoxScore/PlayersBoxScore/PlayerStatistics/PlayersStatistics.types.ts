import type { BoxscorePlayer } from '@nba-app/types-nba';

export type PlayersStatisticsProps = Pick<
  BoxscorePlayer,
  'statistics' | 'played'
>;
