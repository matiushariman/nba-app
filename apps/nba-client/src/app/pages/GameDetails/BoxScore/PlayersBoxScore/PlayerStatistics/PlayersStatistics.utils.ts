export const getPlayingTime = (minutes: string) => {
  const mins = minutes.slice(2, 4);
  const secs = minutes.slice(5, 7);

  return `${mins}:${secs}`;
};

export const getShootingPercentage = (num: number) =>
  `${(num * 100).toFixed(1)}%`;
