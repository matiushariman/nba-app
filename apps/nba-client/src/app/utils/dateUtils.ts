import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
