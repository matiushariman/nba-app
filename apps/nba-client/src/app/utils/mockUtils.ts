export const shouldUseMock = process.env['NODE_ENV'] === 'development';
