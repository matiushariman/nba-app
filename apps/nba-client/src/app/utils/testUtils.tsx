import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
