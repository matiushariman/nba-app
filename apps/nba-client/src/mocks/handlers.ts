import { boxscoreHandlers } from './boxscore/handlers';
import { gamesHandlers } from './games/handlers';

export const handlers = [...boxscoreHandlers, ...gamesHandlers];
