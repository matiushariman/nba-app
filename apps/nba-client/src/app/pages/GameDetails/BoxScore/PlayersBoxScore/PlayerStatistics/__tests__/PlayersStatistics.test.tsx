import { render, screen } from '../../../../../../utils/testUtils';
import PlayerStatistics from '../PlayerStatistics';

import { mockBoxcore } from '../../../../../../../mocks/boxscore/handlers';

const {
  game: {
    awayTeam: { players },
  },
} = mockBoxcore[0];
describe('pages/GameDetails/BoxScore/PlayersBoxScore/PlayersStatistics', () => {
  it('should return DNP if played === 0', () => {
    render(
      <table>
        <tbody>
          <tr>
            <PlayerStatistics played="0" statistics={players[0].statistics} />
          </tr>
        </tbody>
      </table>
    );

    expect(screen.getByText(/dnp/i)).toBeVisible();
  });

  it('should render statistics if played !== 0', () => {
    const { baseElement } = render(
      <table>
        <tbody>
          <tr>
            <PlayerStatistics played="1" statistics={players[0].statistics} />
          </tr>
        </tbody>
      </table>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
