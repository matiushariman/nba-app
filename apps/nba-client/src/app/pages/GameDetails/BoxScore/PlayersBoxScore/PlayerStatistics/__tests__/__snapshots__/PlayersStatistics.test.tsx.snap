// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/BoxScore/PlayersBoxScore/PlayersStatistics should render statistics if played !== 0 1`] = `
<body>
  <div>
    <table>
      <tbody>
        <tr>
          <td
            align="right"
            class="text-sm"
          >
            18:19
          </td>
          <td
            align="right"
            class="text-sm"
          >
            4
          </td>
          <td
            align="right"
            class="text-sm"
          >
            7
          </td>
          <td
            align="right"
            class="text-sm"
          >
            57.1%
          </td>
          <td
            align="right"
            class="text-sm"
          >
            1
          </td>
          <td
            align="right"
            class="text-sm"
          >
            2
          </td>
          <td
            align="right"
            class="text-sm"
          >
            50.0%
          </td>
          <td
            align="right"
            class="text-sm"
          >
            0
          </td>
          <td
            align="right"
            class="text-sm"
          >
            0
          </td>
          <td
            align="right"
            class="text-sm"
          >
            0.0%
          </td>
          <td
            align="right"
            class="text-sm"
          >
            0
          </td>
          <td
            align="right"
            class="text-sm"
          >
            0
          </td>
          <td
            align="right"
            class="text-sm"
          >
            0
          </td>
          <td
            align="right"
            class="text-sm"
          >
            1
          </td>
          <td
            align="right"
            class="text-sm"
          >
            0
          </td>
          <td
            align="right"
            class="text-sm"
          >
            0
          </td>
          <td
            align="right"
            class="text-sm"
          >
            1
          </td>
          <td
            align="right"
            class="text-sm"
          >
            1
          </td>
          <td
            align="right"
            class="text-sm"
          >
            9
          </td>
          <td
            align="right"
            class="text-sm"
          >
            -23
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;
